Yosemite/Yosemite/PointOfSale/PointOfSaleItemService.swift

```diff
@@ -18,42 +13,24 @@ public enum PointOfSaleItemServiceError: Error, Equatable {
 /// Product provider for the Point of Sale feature
 ///
 public final class PointOfSaleItemService: PointOfSaleItemServiceProtocol {
-    private var siteID: Int64
     private let currencyFormatter: CurrencyFormatter
-    private let productsRemote: ProductsRemote
-    private let variationRemote: ProductVariationsRemoteProtocol
-    private let storage: StorageManagerType?
 
-    public init(siteID: Int64,
-                currencySettings: CurrencySettings,
-                network: Network,
-                storage: StorageManagerType? = nil) {
-        self.siteID = siteID
-        self.currencyFormatter = CurrencyFormatter(currencySettings: currencySettings)
-        self.productsRemote = ProductsRemote(network: network)
-        self.variationRemote = ProductVariationsRemote(network: network)
-        self.storage = storage
-    }
+    public var fetchStrategy: PointOfSalePurchasableItemFetchStrategy
```

Comment:
I considered and tried using a let for this. It's doable, but makes it quite clunky to use. There's going to be mutable state somewhere to handle the search term changing, this seems like a reasonable place for it to go, but if you would suggest somewhere else, it'd be good to hear it.

Reply:
> There's going to be mutable state somewhere to handle the search term changing

Do I understand correctly that we need to remember the state so that other methods such as `loadNextItems` or `refreshItems` would behave according to the current strategy?

> If you would suggest somewhere else, it'd be good to hear it.

I don't have a strong opinion. I think I would've picked keeping the state within `Controller`, since it's closer to the view and fetching strategy depends on the view state. We already keep `itemsViewState`. 

Then the `PointOfSaleItemService` methods would take a strategy as a parameter.

Reply:
> Do I understand correctly that we need to remember the state so that other methods such as loadNextItems or refreshItems would behave according to the current strategy?

Yep, exactly. When we set a search strategy, that should be used for subsequent calls.

> I think I would've picked keeping the state within Controller, since it's closer to the view and fetching strategy depends on the view state. We already keep itemsViewState.
>
> Then the PointOfSaleItemService methods would take a strategy as a parameter.

Yeah, I'll give that a try